lib/landable/traffic/event_publisher.rb

```diff
@@ -0,0 +1,101 @@
+module Landable
+  class EventPublisher
+    attr_accessor :page_view, :visit, :event_type
+
+    def initialize(page_view)
+      event_type = ampq_event_mapping[page_view.path]
+      if event_type.is_a?(Hash)
+        request_type = page_view.http_method
+        event_type = event_type[request_type]
+      end
+
+      return unless event_type
+      @page_view = page_view
+
+      @visit = page_view.visit
+      @event_type = event_type
+    end
+
+    def ampq_enabled?
+      @ampq_enabled ||= Landable.configuration.ampq_enabled
+    end
+
+    def ampq_event_mapping
+      @ampq_event_mapping ||= Landable.configuration.ampq_event_mapping
+    end
+
+    def ampq_application_name
+      @ampq_application_name ||= Landable.configuration.ampq_application_name
+    end
+
+    def publish
+      return unless ampq_enabled?
+      MESSAGING_SERVICE.publish(message)
+    end
+
+    def message
+      attribution = visit.attribution
+      referer = visit.referer
+      visitor = visit.visitor
+      user_agent = visitor.try(:raw_user_agent)
+      user_agent_type = user_agent.try(:raw_user_agent_type)
+      {
+          brand: ampq_application_name,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.